### ISSUE TITLE
[BUG] Crash when opening a thread

### ISSUE THREAD

**Issue by K1rakishou** [Adamantcheese/Kuroba#263]
```
D/Kuroba | PageRequestManager: Requesting existing board pages, timeout
E/Kuroba | UNCAUGHT: java.lang.IndexOutOfBoundsException: Index: 34, Size: 35
E/Kuroba | UNCAUGHT: 	at java.util.ArrayList.get(ArrayList.java:437)
E/Kuroba | UNCAUGHT: 	at com.github.adamantcheese.chan.core.presenter.ThreadPresenter.onListScrolledToBottom(ThreadPresenter.java:631)
E/Kuroba | UNCAUGHT: 	at com.github.adamantcheese.chan.ui.layout.ThreadListLayout.lambda$onRecyclerViewScrolled$0$ThreadListLayout(ThreadListLayout.java:164)
E/Kuroba | UNCAUGHT: 	at com.github.adamantcheese.chan.ui.layout.-$$Lambda$ThreadListLayout$wCwyHJXO34_chrEhTvPk0rZTQGs.run(Unknown Source:2)
E/Kuroba | UNCAUGHT: 	at android.os.Handler.handleCallback(Handler.java:883)
E/Kuroba | UNCAUGHT: 	at android.os.Handler.dispatchMessage(Handler.java:100)
E/Kuroba | UNCAUGHT: 	at android.os.Looper.loop(Looper.java:214)
E/Kuroba | UNCAUGHT: 	at android.app.ActivityThread.main(ActivityThread.java:7343)
E/Kuroba | UNCAUGHT: 	at java.lang.reflect.Method.invoke(Native Method)
E/Kuroba | UNCAUGHT: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
E/Kuroba | UNCAUGHT: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:933)
E/Kuroba | UNCAUGHT: .
E/Kuroba | UNCAUGHT: ----------------------------------------
E/Kuroba | UNCAUGHT: END OF CURRENT RUNTIME MESSAGES
E/Kuroba | UNCAUGHT: ----------------------------------------
E/Kuroba | UNCAUGHT: .
E/Kuroba | UNCAUGHT: Android API Level: 29
E/Kuroba | UNCAUGHT: App Version: v4.5.3-dev Unofficial
E/Kuroba | UNCAUGHT: Phone Model: Google Pixel 2 XL
```

**Comment by K1rakishou** [comment]
The thread I opened was scrolled to the bottom before. So it should have opened it at the bottom when crashing. And it only happened once.
Also >Index: 34, Size: 35 
Something is not right here. It should be fine. We have 35 (0...34) elements and we are trying to access the last one (34). Maybe it was changed from a different thread? Probably should check the synchronization of that variable.

**Comment by K1rakishou** [comment]
Gonna be fixed in #247

**Comment by K1rakishou** [comment]
Damn, I thought extracting this into a separate PR because #247 is gonna be postponed at least until the next week but it is impossible to do because there are lots of other changes in that branch. So it's going to be fixed in #247.
